src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/MvcOptions.cs

```diff
@@ -209,6 +213,74 @@ public int MaxModelValidationErrors
             }
         }
 
+        /// <summary>
+        /// Gets or sets the maximum size of a complex collection to model bind. When this limit is reached, the model
+        /// binding system will throw an <see cref="InvalidOperationException"/>.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// When binding a collection, some element binders may succeed unconditionally and model binding may run out
+        /// of memory. This limit constrains such unbounded collection growth.
+        /// </para>
+        /// <para>
+        /// This limit does not <em>correct</em> the bound model. The <see cref="InvalidOperationException"/> instead
+        /// informs the developer of an issue in their model or model binder.
+        /// </para>
+        /// <para>
```

Comment:
I think it's worth mentioning explicitly that this is a safeguard against incorrectly written model binders or problematic models, we expect the developer to fix their code. You hint at this, but I'd like it to be more explict.

Reply:
I'll revise the doc comments and resources in the next iteration